src/Microsoft.ML.Ensemble/Trainer/Binary/EnsembleTrainer.cs

```diff
@@ -0,0 +1,97 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.ML.Runtime;
+using Microsoft.ML.Runtime.CommandLine;
+using Microsoft.ML.Runtime.Data;
+using Microsoft.ML.Runtime.Ensemble;
+using Microsoft.ML.Runtime.Ensemble.OutputCombiners;
+using Microsoft.ML.Runtime.Ensemble.Selector;
+using Microsoft.ML.Ensemble.EntryPoints;
+using Microsoft.ML.Runtime.Internal.Internallearn;
+
+[assembly: LoadableClass(EnsembleTrainer.Summary, typeof(EnsembleTrainer), typeof(EnsembleTrainer.Arguments),
+    new[] { typeof(SignatureBinaryClassifierTrainer), typeof(SignatureTrainer) },
+    EnsembleTrainer.UserNameValue, EnsembleTrainer.LoadNameValue, "pe", "ParallelEnsemble")]
+
+namespace Microsoft.ML.Runtime.Ensemble
+{
+    using TDistPredictor = IDistPredictorProducing<Single, Single>;
+    using TScalarPredictor = IPredictorProducing<Single>;
+    /// <summary>
+    /// A generic ensemble trainer for binary classification.
+    /// </summary>
+    public sealed class EnsembleTrainer : EnsembleTrainerBase<Single, TScalarPredictor,
+        IBinarySubModelSelector, IBinaryOutputCombiner, SignatureBinaryClassifierTrainer>,
+        IModelCombiner<TScalarPredictor, TScalarPredictor>
+    {
+        public const string LoadNameValue = "WeightedEnsemble";
+        public const string UserNameValue = "Parallel Ensemble (bagging, stacking, etc)";
+        public const string Summary = "A generic ensemble classifier for binary classification.";
+
+        public sealed class Arguments : ArgumentsBase
+        {
+            [Argument(ArgumentType.Multiple, HelpText = "Algorithm to prune the base learners for selective Ensemble", ShortName = "pt", SortOrder = 4)]
+            [TGUI(Label = "Sub-Model Selector(pruning) Type",
+                Description = "Algorithm to prune the base learners for selective Ensemble")]
+            public ISupportBinarySubModelSelectorFactory SubModelSelectorType;
+
+            [Argument(ArgumentType.Multiple, HelpText = "Output combiner", ShortName = "oc", SortOrder = 5)]
+            [TGUI(Label = "Output combiner", Description = "Output combiner type")]
+            public ISupportBinaryOutputCombinerFactory OutputCombiner;
+
+            public Arguments()
+            {
+                BasePredictors = new[] { new SubComponent<ITrainer<RoleMappedData, TScalarPredictor>, SignatureBinaryClassifierTrainer>("LinearSVM") };
+                OutputCombiner = new MedianFactory();
+                SubModelSelectorType = new AllSelectorFactory();
```

Comment:
If you are doing it this way, you may as well set them on the fields themselves.

Alternately: you could have had `ArgumentsBase` be generic on the selector and combiner interface types, and continued to use this pattern.

I might prefer the generic way since it will result I *think* in less code overall being written.